LIMS2DB/classes.py

```diff
@@ -211,8 +211,13 @@ def __init__(self, session, log, step):
     def extract_barcode(self, chain):
         barcode=''
         bcp = re.compile("[ATCG\-]{4,}")
+        TENX_PAT = re.compile("SI-(?:GA|NA)-[A-H][1-9][0-2]?")
+        ST_PAT = re.compile("SI-TT-[A-H][1-9][0-2]?")
+        SMARTSEQ_PAT = re.compile('SMARTSEQ[1-9]?-[1-9][0-9]?[A-P]')
         if "NoIndex" in chain:
             return chain
+        if TENX_PAT.match(chain) or TENX_PAT.match(chain) or SMARTSEQ_PAT.match(chain):
```

Comment:
I think there's a typo here, where TENX_PAT is used twice and ST_PAT is missing?

Reply:
Ah, sorry!